editor/src/components/canvas/canvas-strategies/strategies/reparent-helpers/reparent-strategy-helpers.ts

```diff
@@ -54,6 +54,7 @@ export function findReparentStrategies(
   cmdPressed: boolean,
   pointOnCanvas: CanvasPoint,
   allowSmallerParent: AllowSmallerParent,
+  allowWithOnlyTextChildren?: boolean,
```

Comment:
I'd be tempted to make this mandatory, to avoid `undefined` shenanigans.
```suggestion
  allowWithOnlyTextChildren: boolean,
```

Reply:
good idea, but in this specific case I'd approach it from a different direction, looking at where this flag is used, I wonder if there could be a `MetadataUtils.targetElementSupportsChildrenAlsoText` that is same as `MetadataUtils.targetElementSupportsChildren` with `allowWithOnlyTextChildren: true`, but instead of boolean, returns `"supportsChildren' | 'hasOnlyTextChildren' | 'doesNotSupportChildren'`. 
`MetadataUtils.targetElementSupportsChildren` would call this function internally, and return true for supportsChildren, and false otherwise.

the upside of this would be that instead of threading an extra parameter through a bunch of functions, there would be a more general metadata utils function (the hypothetical MetadataUtils.targetElementSupportsChildrenAlsoText), which could be used by the text editing code, and one that could be easily reused by the existing `MetadataUtils.targetElementSupportsChildren`

Reply:
looks amazing!